### ISSUE TITLE
Приложение не реагирует на отключение Wi-Fi

### ISSUE THREAD

**Issue by inoyakaigor** [mosmetro-android/mosmetro-android#191]
Во время подключения отключаю Wi-Fi, но приложение всё равно пытается подключиться. На нажатия на кнопку "Остановить", ни в уведомлении в шторке, ни в окне с логом подключения приложение также не реагирует
Привязка к сети включена, если это важно.

* Highscreen Power Four
* Android 5.1 (почти)чистый

**Comment by SEVENID** [comment]
Подтверждаю заторможенную реакцию на нажатие "остановить". Версия master-#321-70.

**Comment by TheDrHax** [comment]
Скорее всего тут дело в том, что запросы в библиотеке OkHttp иногда намертво зависают. С этим особо ничего не поделать, т.к. на метод `cancel()` такие запросы не реагируют.

Можно попробовать сделать асинхронный HTTP-клиент, который будет создавать отдельный поток и временно блокировать основной. Тогда у нас появится возможность выкидывать зависший поток, чтобы он сам когда-нибудь завершился. Решение не очень, но должно сработать в этих редких случаях.

**Comment by OwnageIsMagic** [comment]
Блокировать юи тред?

**Comment by TheDrHax** [comment]
@OwnageIsMagic Нет, сам алгоритм уже идёт в фоне (IntentService). Просто на каждый запрос запускать AsyncTask и блокировать поток, вызвавший этот запрос до тех пор, пока не придёт прерывание или запрос не завершится. Внешне всё останется также, но можно будет отсоединить зависший запрос и забыть про него.

**Comment by SlimRG** [comment]
Решение не очень, напоминает парадигму watchdogов

**Comment by SlimRG** [comment]
Но если других идей нет...

**Comment by TheDrHax** [comment]
На самом деле я таких зависаний уже год не видел... Может OkHttp поправили?
